packages/cli/src/eventbus/eventBusRoutes.ts

```diff
@@ -0,0 +1,236 @@
+import { eventNamesAll } from './EventMessageClasses';
+import {
+	EventMessageAudit,
+	EventMessageAuditOptions,
+} from './EventMessageClasses/EventMessageAudit';
+import { BadRequestError } from '../ResponseHelper';
+import {
+	MessageEventBusDestinationTypeNames,
+	MessageEventBusDestinationWebhookOptions,
+	EventMessageTypeNames,
+	MessageEventBusDestinationOptions,
+} from 'n8n-workflow';
+
+export const eventBusRouter = express.Router();
+
+// ----------------------------------------
+// TypeGuards
+// ----------------------------------------
+
+const isWithIdString = (candidate: unknown): candidate is { id: string } => {
+	const o = candidate as { id: string };
+	if (!o) return false;
+	return o.id !== undefined;
+};
+
+const isWithQueryString = (candidate: unknown): candidate is { query: string } => {
+	const o = candidate as { query: string };
+	if (!o) return false;
+	return o.query !== undefined;
+};
+
+// TODO: add credentials
+const isMessageEventBusDestinationWebhookOptions = (
+	candidate: unknown,
+): candidate is MessageEventBusDestinationWebhookOptions => {
+	// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+	const o = candidate as MessageEventBusDestinationWebhookOptions;
+	if (!o) return false;
+	// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+	return o.url !== undefined;
+};
+
+const isMessageEventBusDestinationOptions = (
+	candidate: unknown,
+): candidate is MessageEventBusDestinationOptions => {
+	// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+	const o = candidate as MessageEventBusDestinationOptions;
+	if (!o) return false;
+	return o.__type !== undefined;
+};
+
+// ----------------------------------------
+// Events
+// ----------------------------------------
+eventBusRouter.get(
+	`/event`,
+	ResponseHelper.send(async (req: express.Request): Promise<any> => {
+		if (isWithQueryString(req.query)) {
+			switch (req.query.query as EventMessageReturnMode) {
+				case 'sent':
+					return eventBus.getEventsSent();
+				case 'unsent':
+					return eventBus.getEventsUnsent();
+				case 'all':
+				default:
+			}
+		}
+		return eventBus.getEvents();
+	}),
+);
+
+eventBusRouter.post(
+	`/event`,
+	ResponseHelper.send(async (req: express.Request): Promise<any> => {
+		if (isEventMessageOptions(req.body)) {
+			let msg;
+			switch (req.body.__type) {
+				case EventMessageTypeNames.workflow:
+					msg = new EventMessageWorkflow(req.body as EventMessageWorkflowOptions);
+					break;
+				case EventMessageTypeNames.audit:
+					msg = new EventMessageAudit(req.body as EventMessageAuditOptions);
+					break;
+				case EventMessageTypeNames.generic:
+				default:
+					msg = new EventMessageGeneric(req.body);
+			}
+			await eventBus.send(msg);
+		} else {
+			throw new BadRequestError(
+				'Body is not a serialized EventMessage or eventName does not match format {namespace}.{domain}.{event}',
+			);
+		}
+	}),
+);
+
+eventBusRouter.post(
+	`/event/addmany/:count`,
```

Comment:
```suggestion
	`/event/test/addmany/:count`,
```
Considering this is a test endpoint, I think it's worth adding `test` as a URI segment to state the purpose

Reply:
i'll remove it entirely... it wasn't meant to even be there still.